### PR TITLE
fix: add excess_blob_gas for first cancun block

### PR DESCRIPTION
**Pull request by Rjected**
If the excess blob gas is missing, the EVM will otherwise return `ExcessBlobGasNotSet` to the payload builder on the first cancun block, since `next_block_blob_fee` returns `None` for pre-cancun blocks. This sets the excess blob gas to zero on the first cancun block.

### CODE REVIEW

File: crates/payload/builder/src/payload.rs [hunk 1]
```diff
@@ -177,9 +177,24 @@ impl PayloadBuilderAttributes {
...
+
         // ensure we're not missing any timestamp based hardforks
         cfg.spec_id = revm_spec_by_timestamp_after_merge(chain_spec, self.timestamp);
 
+        // if the parent block did not have excess blob gas (i.e. it was pre-cancun), but it is
+        // cancun now, we need to set the excess blob gas to the default value
+        let excess_blob_gas = parent.next_block_blob_fee().map_or_else(
+            || {
+                if chain_spec.is_cancun_activated_at_timestamp(self.timestamp) {
+                    // default excess blob gas is zero
```
Comment (mattsse) [inline crates/payload/builder/src/payload.rs:189]: I think this can be simplified by checking the spec_id instead?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/payload/builder/src/payload.rs (hunk 1 above)
Comment (Rjected) [reply crates/payload/builder/src/payload.rs:189]: oh right, I didn't think of that

File: crates/payload/builder/src/payload.rs [hunk 2]
```diff
@@ -177,9 +177,24 @@ impl PayloadBuilderAttributes {
...
         cfg.chain_id = chain_spec.chain().id();
+
         // ensure we're not missing any timestamp based hardforks
         cfg.spec_id = revm_spec_by_timestamp_after_merge(chain_spec, self.timestamp);
 
+        // if the parent block did not have excess blob gas (i.e. it was pre-cancun), but it is
+        // cancun now, we need to set the excess blob gas to the default value
+        let excess_blob_gas = parent.next_block_blob_fee().map_or_else(
+            || {
+                if cfg.spec_id == SpecId::CANCUN {
```
Comment (mattsse) [inline crates/payload/builder/src/payload.rs:188]:
```suggestion
                if cfg.spec_id >= SpecId::CANCUN {
```

**Review by mattsse — APPROVED** [review]

File: crates/payload/builder/src/payload.rs (hunk 2 above)
Comment (mattsse) [reply crates/payload/builder/src/payload.rs:188]: I think >= would cover all cases were additional forks are activated at block 0, but probably not a realistic edge case so not sure if even useful to check

Comment (mattsse) [reply crates/payload/builder/src/payload.rs:188]: in this case the genesis block should have it as 0 so we don't need to check this

**Pull request merged**